napari/components/add_layers_mixin.py

```diff
@@ -155,91 +159,77 @@ def add_image(
         elif data is None:
             data = io.magic_imread(path)
 
+        # doing this here for IDE/console autocompletion in add_image function.
+        kwargs = {
+            'rgb': rgb,
+            'is_pyramid': is_pyramid,
+            'colormap': colormap,
+            'contrast_limits': contrast_limits,
+            'gamma': gamma,
+            'interpolation': interpolation,
+            'rendering': rendering,
+            'iso_threshold': iso_threshold,
+            'attenuation': attenuation,
+            'name': name,
+            'metadata': metadata,
+            'scale': scale,
+            'translate': translate,
+            'opacity': opacity,
+            'blending': blending,
+            'visible': visible,
+        }
+
         if channel_axis is None:
-            if colormap is None:
-                colormap = 'gray'
-            if blending is None:
-                blending = 'translucent'
-            layer = layers.Image(
-                data,
-                rgb=rgb,
-                is_pyramid=is_pyramid,
-                colormap=colormap,
-                contrast_limits=contrast_limits,
-                gamma=gamma,
-                interpolation=interpolation,
-                rendering=rendering,
-                iso_threshold=iso_threshold,
-                attenuation=attenuation,
-                name=name,
-                metadata=metadata,
-                scale=scale,
-                translate=translate,
-                opacity=opacity,
-                blending=blending,
-                visible=visible,
-            )
-            self.add_layer(layer)
-            return layer
+            kwargs['colormap'] = kwargs.get('colormap', 'gray')
+            kwargs['blending'] = kwargs.get('blending', 'translucent')
```

Comment:
kwargs.get will return the default value when the key doesn't exist, not when it exists but the value is None. Since the key *does* exist, this is the wrong construct here. I think?? Ditto for other calls to `.get`

Reply:
it's funny... I recently made your exact same point to someone in an external PR, but somehow forgot it here!  I meant to do `kwargs['colormap'] or 'gray'`, that ok?  The only difference from the current code is that it also excludes the empty string, which in this context is as useless as `None`